First-Peoples-Cultural-Council/fv-be
# FW-4433 Add unittests to search API

### Description of Changes
- Added translation field in search document.
- Added sub-queries in the query builder.
- Added more tests for different cases, and checking all subqueries are present.
- Search API documentation has been updated.

### Checklist
- [x] README / documentation has been updated if needed ([wiki link](https://firstvoices.atlassian.net/wiki/spaces/FIR/pages/edit-v2/288620545?draftShareId=22d224c8-6c46-4cb1-bcdc-59206d4bbe78&inEditorTemplatesPanel=auto_closed))
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- ~Admin Panel has been updated if models have been added or modified~
- ~Migrations have been updated and committed if applicable~

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [x] Pull the branch and test locally

### Additional Notes
To test locally:

1. Ensure that elastic-search server is running locally. Instructions can be found in the readme to get the docker container running.
2. Add any new words and they will be indexed automatically. Add and remove translations and verify that they can also be searched. Wildcard matching is not present as of now, but fuzzy matching should work.
3. The following url can be used with the q parameter
`url: http://localhost:8000/api/1.0/search/?q=wordTitle`
`q: enter a search query which can be matched with the title or translation field in the dictionaryEntry model as of now.`
4. Verify that the search results returned contains the query param either in the document title or the translation text.
